plugins/CoreAdminHome/Commands/PurgeOldArchiveData.php

```diff
@@ -58,6 +58,8 @@ protected function configure()
         $this->addOption('exclude-invalidated', null, InputOption::VALUE_NONE, "Do not purge invalidated archive data.");
         $this->addOption('exclude-ranges', null, InputOption::VALUE_NONE, "Do not purge custom ranges.");
         $this->addOption('skip-optimize-tables', null, InputOption::VALUE_NONE, "Do not run OPTIMIZE TABLES query on affected archive tables.");
+        $this->addOption('include-year-archives', null, InputOption::VALUE_NONE, "If supplied, the command will purge archive tables that contain year archives for every supplied date.");
```

Comment:
For temporary archive purging, the scheduled task goes through every archive table, so year archives are affected.

For invalidated archive purging, the scheduled task uses the ArchivesToPurgeDistributedList, so if the year archive is invalidated correctly, the year table should get purged. I'm not very familiar w/ the invalidation code, though, just purging.